EIPS/eip-7620.md

```diff
@@ -0,0 +1,266 @@
+
+## Motivation
+    
+This EIP uses terminology from the [EIP-3540](./eip-3540.md) which introduces the EOF format.
+
+EOF aims to remove code observability, which is a prerequisite to legacy EVM contract creation logic using create transactions, `CREATE` and `CREATE2`, because both the initcode and code are available to the EVM and can be manipulated. On the same premise, EOF removes opcodes like `CODECOPY` and `EXTCODECOPY`, introducing EOF subcontainers as a replacement to cater for factory contracts creating other contracts.
+    
+The new instructions and the new transaction type introduced in this EIP operate on EOF containers enabling all the use cases of contract creation that legacy EVM has.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+    
+We introduce three new instructions on the same block number [EIP-3540](./eip-3540.md) is activated on:
+
+1. `CREATE3` (`0xec`)
+2. `CREATE4` (`0xed`)
+3. `RETURNCONTRACT` (`0xee`)
+    
+### Transaction Types
+
+Introduce new transaction type `InitcodeTransaction` which extends [EIP-1559](./eip-1559.md) (type 2) transaction by adding a new field `initcodes: List[ByteList[MAX_INITCODE_SIZE], MAX_INITCODE_COUNT]`.
+
+The `initcodes` can only be accessed via the `CREATE4` instruction (see below), therefore `InitcodeTransactions` are intended to be sent to contracts including `CREATE4` in their execution.
+
+We introduce a standardised Creator Contract (i.e. written in EVM, but existing at a known address, such as precompiles), which eliminates the need to have create transactions with empty `to`. Deployment of the Creator Contract will require an irregular state change at EOF activation block. Note that such introduction of the Creator Contract is needed, because only EOF contracts can create EOF contracts. See below for Creator Contract code.
+
+Under transaction validation rules `initcodes` are not validated for conforming to the EOF specification. They are only validated when accessed via `CREATE4`. This avoids potential DoS attacks of the mempool. If during the execution of an `InitcodeTransaction` no `CREATE4` instruction is called, such transaction is still valid.
+
+`initcodes` data is similar to calldata for two reasons:
+
+1. It must be fully transmitted in the transaction.
+2. It is accessible to the EVM, but it can't be fully loaded into EVM memory.
+
+For these reason `initcodes` data costs the same cost as for calldata (16 gas for non-zero bytes, 4 for zero bytes -- see [EIP-2028](./eip-2028.md)).
```

Comment:
can we create a small section with a heading GAS SCHEDULE or somthing similar for better clarity and readability?